staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -42,6 +43,14 @@ func WithAudit(handler http.Handler, sink audit.Sink, policy policy.Checker, lon
 	if sink == nil || policy == nil {
 		return handler
 	}
+
+	// We want to track how much time a request spends in audit filter.
```

Comment:
I initially though that having this kind of wrapper is overkill and preferred that to have more custom to the P&F.

But after thinking more about that and seeing this PR, I actually like the genericity of it and the fact we can use it for other filters too.
@lavalamp - does that seem reasonable to you too?

Reply:
Hi @lavalamp @wojtek-t, I have enabled it for `priority and fairness`, `authentication` `authorization`, `impersonation` and `audit` filters for now.

Reply:
shouldn't we have this in the default handler chain builder func instead? The fan out would be smaller than adding it in all filters.

Reply:
@sttts @MikeSpreitzer I have moved the tracking inside the `DefaultBuildHandlerChain` function now.